packages/core/src/utils/metadata-utils/pointcloud-metadata-utils.ts

```diff
@@ -0,0 +1,319 @@
+import { PointConfig, RefineStrategy } from '@tiledb-inc/viz-common';
+import {
+  AssetOptions,
+  PointDataContent,
+  SceneOptions,
+  TillingScheme
+} from '../../types';
+import { PointCloudMetadata } from '../../tile';
+import getTileDBClient from '../getTileDBClient';
+import { deserializeBuffer, get3DTransformedBoundingInfo } from './utils';
+import { Vector3 } from '@babylonjs/core';
+import { Matrix } from 'mathjs';
+import { encode3D } from '@tiledb-inc/viz-common';
+import { PointTileContent } from '../../tile/model/point/pointContent';
+import { Tile } from '../../tile/model/tile';
+import { Attribute } from '@tiledb-inc/viz-common';
+import { Domain } from '@tiledb-inc/viz-common';
+import { FeatureType } from '@tiledb-inc/viz-common';
+import { Feature } from '@tiledb-inc/viz-common';
+import proj4 from 'proj4';
+import { getQueryDataFromCache, writeToCache } from '../cache';
+import {
+  ArraySchema,
+  GroupContents,
+  GroupInfo
+} from '@tiledb-inc/tiledb-cloud/lib/v1';
+
+export async function getPointCloudMetadata(
+  options: AssetOptions,
+  config?: PointConfig,
+  sceneOptions?: SceneOptions
+): Promise<PointCloudMetadata> {
+  const client = getTileDBClient({
+    ...(options.token ? { apiKey: options.token } : {}),
+    ...(options.tiledbEnv ? { basePath: options.tiledbEnv } : {})
+  });
+
+  if (!options.pointGroupID) {
+    throw new Error('Point group ID is undefined');
+  }
+
+  let info: GroupInfo = await getQueryDataFromCache(
+    options.pointGroupID,
+    'info'
+  );
+  let members: GroupContents = await getQueryDataFromCache(
+    options.pointGroupID,
+    'members'
+  );
+  let groupMetadata = await getQueryDataFromCache(
+    options.pointGroupID,
+    'groupMetadata'
+  );
+
+  if (!info || !members || !groupMetadata) {
+    [info, members, groupMetadata] = await Promise.all([
+      client.groups.API.getGroup(options.namespace, options.pointGroupID).then(
+        x => x.data
+      ),
+      client.groups.getGroupContents(options.namespace, options.pointGroupID),
+      client.groups.V2API.getGroupMetadata(
+        options.namespace,
+        options.pointGroupID
+      )
+        .then((response: any) => response.data.entries ?? [])
+        .then((data: any) => {
+          return data.reduce((map: any, obj: any) => {
+            map[obj.key] = deserializeBuffer(obj.type, obj.value);
+            return map;
+          }, {});
+        })
+    ]);
+
+    await writeToCache(options.pointGroupID, 'info', info);
```

Comment:
you could make these `'info'`, `'members'`, `'groupMetadata'` const names at top of file, just a thought ... really this defines a schema in some sense ...

Reply:
@XanthosXanthopoulos ping 🙏

Reply:
The assignment of these variable may happen at two different places depending on whether or not exists cached versions